packages/host/app/components/ai-assistant/panel.gts

```diff
@@ -379,21 +387,25 @@ export default class AiAssistantPanel extends Component<Signature> {
     }
   }
 
-  private get currentRoom() {
+  @cached
+  private get roomResources() {
+    return this.matrixService.roomResources;
+  }
+
+  private get roomResource() {
     return this.currentRoomId
-      ? this.roomResources.get(this.currentRoomId)?.room
+      ? this.roomResources.get(this.currentRoomId)
       : undefined;
   }
 
-  @cached
-  private get roomResources() {
-    return this.matrixService.roomResources;
+  private get roomModel() {
+    return this.roomResource ? this.roomResource.room : undefined;
   }
 
   private loadRoomsTask = restartableTask(async () => {
     await this.matrixService.flushMembership;
     await this.matrixService.flushTimeline;
-    await Promise.all([...this.roomResources.values()].map((r) => r.loading));
+    await Promise.all([...this.roomResources.values()].map((r) => r.loading)); //does this even do anything?
```

Comment:
regarding that comment, yes it absolute does something 🙂. This will await all the room resources to be finished loadg. without that `await` you might have a room resource that is still loading. I looks like `enterRoomInitially()` wants all the async resolved since it is synchronous.